src/rootfstreemanager.cc

```diff
@@ -55,7 +55,12 @@ RootfsTreeManager::RootfsTreeManager(const PackageConfig& pconfig, const Bootloa
       http_client_{http},
       gateway_url_{pconfig.ostree_server},
       keys_{keys},
-      cfg_{pconfig} {}
+      cfg_{pconfig} {
+  // TODO: Move this logic to the Sysroot ctor, it requires moving the watermark parameter to it to.
+  OSTree::Repo repo{sysroot_->repoPath()};
+  repo.setStorageUsageWatermark(cfg_.SysrootStorageWatermark);
+  sysroot_->reload();
```

Comment:
Two configuration knobs for the same storage parameter (high watermark of storage usage)?

@mike-scott Maybe one know will suffice?

It's easier to shoot yourself in the foot with two than with one :).

Reply:
Maybe the second one could be just boolean that determines whether to overwrite the watermark set in the ostree config with the one set in the aklite config sota.toml?

Reply:
perhaps - that's what we are trying to protect - someone could have already set this value in their image. So if we just start setting it for them automagically - we overwrite their work.

Reply:
So, by default, `sysroot_storage_watermark` defined in the aklite config is applied only to the delta pull. The value set in the ostree config (default 3%) is applied to the non-delta case.
A user can override the ostree config default value in the custom recipe at LmP level. Or optionally, set the additionally boolean param instructing to apply `sysroot_storage_watermark` to the non-delta too.

Reply:
> someone could have already set this value in their image. So if we just start setting it for them automagically - we overwrite their work.

Yep, I thought about it too, some devices may might already exceeded the default 90% but less than 97%, so they are still updatable. This change will make those devices unupdatable. 

Hence, the boolean may be better, because we might confuse customers with three values of storage watermark (defined in ostree config + two new knobs).

Reply:
I added a new parameter `min_free_space_percent` that allows to override/set `core.min-free-space-percent` value in the ostree config.

@doanac @mike-scott Please, let me know if this is what you expected. If so, I'll tidy up commits then.